core/src/main/java/org/kiwix/kiwixmobile/core/main/KiwixTextToSpeech.java

```diff
@@ -131,23 +132,14 @@ public void readAloud(WebView webView) {
           context.getResources().getString(R.string.tts_lang_not_supported),
           Toast.LENGTH_LONG).show();
       } else {
-        // check if voice needs to be installed for API below 21
-        if(VERSION.SDK_INT < VERSION_CODES.LOLLIPOP
-          && tts.getFeatures(locale).contains(TextToSpeech.Engine.KEY_FEATURE_NOT_INSTALLED)){
-          Toast.makeText(context,
-            context.getResources().getString(R.string.tts_lang_not_supported),
-            Toast.LENGTH_LONG).show();
-          return;
-        }
-
         tts.setLanguage(locale);
 
-        // check if voice needs to be installed for API 21 or higher
-        if (VERSION.SDK_INT >= VERSION_CODES.LOLLIPOP
-          && tts.getVoice().getFeatures().contains(TextToSpeech.Engine.KEY_FEATURE_NOT_INSTALLED)){
-          Toast.makeText(context,
-            context.getResources().getString(R.string.tts_lang_not_supported),
-            Toast.LENGTH_LONG).show();
+        if ((VERSION.SDK_INT < VERSION_CODES.LOLLIPOP // check if voice needs to be installed for API below 21
```

Comment:
I was envisaging a more condensed version as you are still repeating the condition check, you can do it in a method because inline could look a little clunky, pseudocode:
``` 
getFeatures(){
VERSION.SDK_INT >= VERSION_CODES.LOLLIPOP ? tts.getVoice.getFeatures() : tts.getFeatures(locale)
}
```
then the condition looks much nicer and we don't need comments
```
if(getFeatures(tts, locale).contains(TextToSpeech.Engine.KEY_FEATURE_NOT_INSTALLED))
```

Reply:
So just to be sure I understand you, I will add a private method to this class called `getFeatures()` and query this method for the required feature, right ?

Reply:
yes please, I skipped over adding some parameters for brevity

Reply:
I am sorry for the late commit because it's the weekend where I am from. I added the method but the linter refused commiting the inline ternary operation `VERSION.SDK_INT >= VERSION_CODES.LOLLIPOP ? tts.getVoice.getFeatures() : tts.getFeatures(locale)` even though it builds and works, so I switched to an if condition, please tell me if it could be beautified even more

Reply:
Why did the linter complain? It usually does so for a very good reason. I would prefer the ternary

Reply:
I got those results from the Android studio
![image](https://user-images.githubusercontent.com/34913464/74851506-02ea1c00-5344-11ea-8e5b-d91c34a41d43.png)
![image](https://user-images.githubusercontent.com/34913464/74851620-290fbc00-5344-11ea-9846-1b5ac77fc374.png)
![image](https://user-images.githubusercontent.com/34913464/74851732-4e9cc580-5344-11ea-83ea-a3f12d088df4.png)

Reply:
So put your cursor on the red underlined methods and press `ctrl+enter`, you will get the option to add the annotation. You have guarded your access with a conditional so lint is being a bit silly here. Use a suppression annotation because lint is just being stupid.
![image](https://user-images.githubusercontent.com/3358282/74919048-e68cc480-53c1-11ea-93d6-536287dae351.png)

Reply:
Alright, will be pushing soon. Should I worry about the code coverage check failing ?

Reply:
as long as it fails here:
![image](https://user-images.githubusercontent.com/3358282/75016877-d9d0a500-5483-11ea-9d79-e32c471d4e75.png)
it is fine. We need to use some credentials to upload our coverage reports and since this PR comes from a fork unfortunately you do not have access to those credentials